apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -2,21 +2,48 @@ defmodule GameBackend.Matches do
   @moduledoc """
   Matches
   """
+  alias GameBackend.Users
+  alias GameBackend.Users.Currencies
   alias Ecto.Multi
   alias GameBackend.Repo
   alias GameBackend.Matches.ArenaMatchResult
 
-  def create_arena_match_results(match_id, results) do
-    transaction = Multi.new()
+  def create_arena_match_results(results) do
+    {:ok, conrrency_config_json} =
+      Application.app_dir(:game_backend, "priv/currencies_rules.json")
+      |> File.read()
 
-    transaction =
-      Enum.reduce(results, transaction, fn result, transaction_acc ->
-        attrs = Map.put(result, "match_id", match_id)
-        changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, attrs)
-        transaction_acc = Multi.insert(transaction_acc, {:insert, attrs["user_id"]}, changeset)
-        transaction_acc
-      end)
+    currency_config =
+      Jason.decode!(conrrency_config_json)
```

Comment:
Having to read and JSON parse on every request is not ideal, if we could read and process this config during compilation (in `config.exs`) or at startup we could have a nicer presentation and save some milliseconds on this

For example, we could end up saving the config as a map like
```elixir
%{
  :distributions => [50, 250, 9999],
  50 => %{1 => 10, 2 => 8, ...},
  250 => %{1 => 8, 2 => 6, ...},
  9999 => %{1 => 1, 2 => -4, ...},
  :infinity => %{1 => 0, 2 => -1, ...},
}
 ```
 
 So we have a `distributions` entry that lists the max amounts, then a key with that distribution which has a map where every position is a key. Also, we have the special case `:infinity` to handle distributions after whatever max amount we set

Reply:
Yeah i didn't quite liked parsing the config file on every request, i didn't though about add it to the config file thx! i moved this to the `runtime.exs` file and we can simply read it from there! i keep the old system format tho do you think we should change it?